pandas/tools/pivot.py

```diff
@@ -436,18 +446,71 @@ def crosstab(index, columns, values=None, rownames=None, colnames=None,
     data.update(zip(rownames, index))
     data.update(zip(colnames, columns))
 
+    if values is None and aggfunc is not None:
+        raise ValueError("aggfunc cannot be used without values.")
+
     if values is None:
         df = DataFrame(data)
         df['__dummy__'] = 0
         table = df.pivot_table('__dummy__', index=rownames, columns=colnames,
                                aggfunc=len, margins=margins, dropna=dropna)
-        return table.fillna(0).astype(np.int64)
+        table = table.fillna(0).astype(np.int64)
+
     else:
         data['__dummy__'] = values
         df = DataFrame(data)
         table = df.pivot_table('__dummy__', index=rownames, columns=colnames,
                                aggfunc=aggfunc, margins=margins, dropna=dropna)
-        return table
+
+    # Post-process
+    table = _normalize(table, normalize=normalize, margins=margins,
+                       aggfunc=aggfunc)
+
+    return table
+
+
+def _normalize(table, normalize, margins, aggfunc):
+
+    if aggfunc is not None and normalize is not False:
```

Comment:
I think moving this check to ``crosstab`` and removing ``aggfunc`` from ``_normalize`` is clearer.